se/typography.py

```diff
@@ -532,25 +537,47 @@ def convert_british_to_american(xhtml: str) -> str:
 	"""
 	xhtml = regex.sub(r"“", r"<ldq>", xhtml)
 	xhtml = regex.sub(r"”", r"<rdq>", xhtml)
+	# mark as apostrophe reversed (opening rather than closed) quote on known dialect;
+	# at, is, and the decades are also words on their own, but are deemed more likely to
+	# be elisions than starting a quote
+	xhtml = regex.sub(r"([^A-Za-z])‘([Aa]lf|at|[Aa]ve|[Ee]|[Ee]ard|[Ee]lp(ed|s)?|[Ee]m|[Ee]re|[Ee]rself|[I]im|is|[Ii]sself|[Ii]story|[Oo]ller|[Oo]w|[Oo]wever|[Rr]e|(twen|thir|four|fif|six|seven|eigh|nine)ties)\b",r"\1<ap>\2", xhtml)
+	# treat any remaining opening single quotes as lsq
 	xhtml = regex.sub(r"‘", r"<lsq>", xhtml)
-	xhtml = regex.sub(r"<rdq>⁠ ’(\s+)", r"<rdq> <rsq>\1", xhtml)
-	xhtml = regex.sub(r"<rdq>⁠ ’</", r"<rdq> <rsq></", xhtml)
-	xhtml = regex.sub(r"([\.\,\!\?\…\:\;])’", r"\1<rsq>", xhtml)
-	xhtml = regex.sub(r"—’(\s+)", r"—<rsq>\1", xhtml)
-	xhtml = regex.sub(r"—’</", r"—<rsq></", xhtml)
-	xhtml = regex.sub(r"([\p{Lowercase_Letter}])’([\p{Lowercase_Letter}])", r"\1<ap>\2", xhtml)
-	xhtml = regex.sub(r"(\s+)’([\p{Lowercase_Letter}])", r"\1<ap>\2", xhtml)
+	# ’a’ is two apostrophes
+	xhtml = regex.sub(r"(\W)’a’(\W)", r"\1<ap>a<ap>\2", xhtml)
+	# o’ on its own is an apostrophe
+	xhtml = regex.sub(r"\bo’(\W)", r"o<ap>\1", xhtml)
+	# treat any instance of a closing single quote followed by a letter as an apostrophe
+	xhtml = regex.sub(r"’([[:alpha:]])", r"<ap>\1", xhtml)
```

Comment:
`[:alpha:]` only matches `a-zA-Z` excluding letters with diacritics, non-Latin, etc. You should use `[\p{Letter}]` instead to match the Unicode class of all letters.

Reply:
I have not found that to be true, and didn't here. They include all Unicode alpha in the editors I use/have used, and I just tested this script on `Will b2a identify ’ís and ’Ès as an apostrophe?` and it worked (I commented out everything after that line, so I could see that it tagged them as <op>). The regex docs also specify they are equivalent. I find the Posix character classes easier to read (they're more common in my experience) and easier to test in an editor (no translation needed), and I've never had any issue with Unicode characters. I have definitely never found it the same as [A-Za-z]. My understanding has been that's why those classes exist in the first place, to provide a Unicode equivalent to the various "ascii" character classes ([A-Z], [0-9], etc.).

Reply:
Well, in any case the rest of the codebase uses Unicode classes in regex so we should do that  here too for consistency.

Reply:
Changed, but I was going to suggest the opposite (before this conversation), for the reasons stated above. The Posix classes do the same thing, are more commonly used, and easier to copy/paste into a text editor and use. They also eliminate the double curly bracket problem in typogrify.

Reply:
We use a variety of classes like `\p{Lowercase_letter}`, etc. throughout the codebase, therefore I would rather be consistent in uses classes everywhere, instead of using classes somewhere and unrelated aliases elsewhere.

Reply:
I'm not advocating a mixture, I'm advocating that we should use posix classes throughout, for the reasons stated.

Reply:
Are there a posix class that is equivalent to `\p{script=Latin}`, `\p{Lowercase_letter}`, `\p{Punctuation}`, `\p{M}`, etc.?

Reply:
[This](https://www.regular-expressions.info) is one of the best resources I've found for regex in general, and [this page](https://www.regular-expressions.info/posixbrackets.html) in particular lists the posix classes and their unicode equivalents.